src/main/scala/com/gu/editorial/permissions/client/PermissionsStore.scala

```diff
@@ -57,21 +57,29 @@ class PermissionsStore(provider: Option[PermissionsStoreProvider] = None)(implic
 
 trait PermissionsStoreProvider {
   val store: Agent[PermissionsStoreModel]
+  def storeIsEmpty: Boolean
 }
 
 
-private[client] final class PermissionsStoreFromS3(refreshFrequency: Option[FiniteDuration] = Some(Duration(1, MINUTES)),
-                                                s3Client: Option[AmazonS3] = None)
-                                               (implicit config: PermissionsConfig,
-                                                actorSystem: ActorSystem = ActorSystem(),
-                                                executionContext: ExecutionContext) extends PermissionsStoreProvider {
+private[client] final class PermissionsStoreFromS3(config: PermissionsConfig,
+                                                   refreshFrequency: Option[FiniteDuration] = Some(Duration(1, MINUTES)),
+                                                   s3Client: Option[AmazonS3] = None)
+                                                  (implicit actorSystem: ActorSystem = ActorSystem(),
+                                                   executionContext: ExecutionContext) extends PermissionsStoreProvider {
 
   implicit lazy val logger = LoggerFactory.getLogger(getClass)
 
   implicit private val timeout = Timeout(Duration(5, SECONDS))
 
   val store: Agent[PermissionsStoreModel] = Agent(PermissionsStoreModel.empty)
 
+  def storeIsEmpty = {
+    store.get() match {
+      case PermissionsStoreModel.empty => true
+      case s: PermissionsStoreModel => false
+    }
+  }
+
   private val s3 = s3Client.getOrElse {
```

Comment:
You could expose an 'init' function which speaks to S3 and either returns the contents of the bucket or an error on first call - that way you don't need to handle the case where the cache just isn't populated yet. Up to you though